rosbag2_storage/include/rosbag2_storage/message_definition.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023, Foxglove Technologies.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_STORAGE__MESSAGE_DEFINITION_HPP_
+#define ROSBAG2_STORAGE__MESSAGE_DEFINITION_HPP_
+
+#include <string>
+#include <cassert>
+
+#include "rosbag2_storage/visibility_control.hpp"
+
+namespace rosbag2_storage
+{
+
+struct MessageDefinition
+{
+  /// @brief  The name of the type.
+  ///
+  /// Should match the `name` in TopicMetadata for all topics using this message definition.
+  std::string type_name;
```

Comment:
@james-rms It seems there are a big confusion about this `type_name` fileld.
After some investigation I think that it really shall be **topic_type** and shall correspond to the field `type` from `TopicMetada` struct.  
We don't use this field on storage plugin level since we need `TopicMetada::type` instead. 
Although on all upper levels we treat it as it mapped to the `TopicMetada::type` instead of the `TopicMetada::name`. Take a look on all tests and even in recorder.cpp where we are calling get_full_text(..)